File: examples/event/publish/node/publishAnEvent.js
const { EventService } = require("m3o/event");

// Publish a event to the event stream.
async function publishAnEvent() {
  let eventService = new EventService(process.env.MICRO_API_TOKEN);
  let rsp = await eventService.publish({
    message: {
      id: "1",
      type: "signup",
      user: "john",
    },
    topic: "user",
  });
  console.log(rsp);
}

publishAnEvent();

File: examples/event/publish/go/publishAnEvent.go
package example

import (
	"fmt"
	"os"

	"github.com/micro/services/clients/go/event"
)

// Publish a event to the event stream.
func PublishAnEvent() {
	eventService := event.NewEventService(os.Getenv("MICRO_API_TOKEN"))
	rsp, err := eventService.Publish(&event.PublishRequest{
		Message: map[string]interface{}{
			"type": "signup",
			"user": "john",
			"id":   "1",
		},
		Topic: "user",
	})
	fmt.Println(rsp, err)
}
